ml_inference_offloading/src/androidTest/java/ai/nnstreamer/ml/inference/offloading/LocalDatabaseTest.kt

```diff
@@ -0,0 +1,70 @@
+package ai.nnstreamer.ml.inference.offloading
+
+import ai.nnstreamer.ml.inference.offloading.data.AppDatabase
+import ai.nnstreamer.ml.inference.offloading.data.Model
+import ai.nnstreamer.ml.inference.offloading.data.ModelDao
+import ai.nnstreamer.ml.inference.offloading.data.OffloadingService
+import ai.nnstreamer.ml.inference.offloading.data.OffloadingServiceDao
+import android.content.Context
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import org.junit.Assert.*
+import org.junit.Before
+import org.nnsuite.nnstreamer.Pipeline
+import java.io.IOException
+import kotlin.jvm.Throws
+
+@RunWith(AndroidJUnit4::class)
+class LocalDatabaseTest {
+    private lateinit var database: AppDatabase
+    private lateinit var modelDao: ModelDao
+    private lateinit var offloadingServiceDao: OffloadingServiceDao
+
+    @Before
+    fun createDb() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        database = Room.inMemoryDatabaseBuilder(
+            context, AppDatabase::class.java
+        ).build()
+        modelDao = database.modelDao()
+        offloadingServiceDao = database.offloadingServiceDao()
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        database.close()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun writeOffloadingServiceAndReadInList() = runTest {
+        val modelId = 1
+        val model = Model(modelId, "model_name")
+        modelDao.insert(model)
+
+        val serviceId = 1
+        val port = 3030
+        val offloadingService = OffloadingService(serviceId, modelId, port, Pipeline.State.NULL)
+        offloadingServiceDao.insert(offloadingService)
+
+        launch {
+            val byId = offloadingServiceDao.getOffloadingService(1)
+            byId.take(1).collect {
```

Comment:
```suggestion
            val byId = offloadingServiceDao.getOffloadingService(serviceId)
            try {
                byId.first().collect {
                ...
            } catch (e: NoSuchElementException) {
               ...
            }
```

Reply:
Updated!
Thanks to this, I discovered that the dao modification was not reflected because the instrumented test was only run on the device.